src/main/webapp/one-off/databaseInit.java

```diff
@@ -40,104 +39,172 @@
 import javax.servlet.http.HttpServletResponse;
-import javax.servlet.ServletException;
-import java.util.*;
-import java.io.*;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Paths;
-import java.nio.file.Files;
+import org.apache.commons.lang3.StringUtils;
 
-
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/test")
 public class TestUploadServlet extends HttpServlet {
+  private String RECIPES_DIRECTORY = "RECIPES_DIRECTORY";
+  private String RECIPE_FILE_EXTENSION = ".txt";
+
+  /**
+   * Receives a recipe's properties and creates an entity, for the database and a document for the index
+   */
+  public void upload(
+    String title,
+    String imgURL,
+    ArrayList<String> ingredients,
+    ArrayList<String> stepList
+  )
+    throws Exception, InterruptedException, IOException {
+    IndexSpec indexSpec = IndexSpec
+      .newBuilder()
+      .setName("recipesIndex")
+      .build();
+    Index index = SearchServiceFactory.getSearchService().getIndex(indexSpec);
 
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     /**
-      * upload takes the several properties of a recipes, uses them in creating a new recipe entity and upload thi entity to the database
-     **/
-    public void upload(String  title, String imgURL, ArrayList<String> ingredients, ArrayList<String> stepList) throws Exception, IOException {
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    
-        Entity recipeEntity = new Entity("Recipe");
-
-        recipeEntity.setProperty("title", title);
-        recipeEntity.setProperty("index_title", title.toLowerCase());
-        recipeEntity.setProperty("imgURL", imgURL);
-        recipeEntity.setProperty("ingredients", ingredients);
-        recipeEntity.setProperty("stepList", stepList);
-        recipeEntity.setProperty("author", "Piece of Cake");
-        recipeEntity.setProperty("difficulty", "N/A");
-        recipeEntity.setProperty("prep_time", "N/A");
-        recipeEntity.setProperty("cook_time", "N/A");
-
-        datastore.put(recipeEntity);
-    }
+     * We alocate an id here because we need it inside the index documents
+     * This will help us to search for the recipes in an efficient way
+     */
+    KeyRange keyRange = datastore.allocateIds("Recipe", 1L);
+
+    Entity recipeEntity = buildRecipeEntity(
+      keyRange,
+      title,
+      imgURL,
+      ingredients,
+      stepList
+    );
+    Document recipeDocument = buildRecipeDocumentForIndexing(recipeEntity);
+
+    datastore.put(recipeEntity);
+    index.put(recipeDocument);
+  }
+
+  /**
+   * Reads the file given as parameter as a list of strings
+   */
+
+  public static List<String> readFileInList(String fileName)
+    throws Exception, IOException {
+    List<String> lines = Collections.emptyList();
+    lines = Files.readAllLines(Paths.get(fileName), StandardCharsets.UTF_8);
+    return lines;
```

Comment:
Files.readAllLines is already returning a List<String>, you can do

return Files.readAllLines(Paths.get(fileName), StandardCharsets.UTF_8)